Shared/AssertionUtils.swift

```diff
@@ -1,14 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-import Foundation
-
-/**
- Assertion for checking that the call is being made on the main thread.
-
- - parameter message: Message to display in case of assertion.
- */
-public func assertIsMainThread(_ message: String) {
-    assert(Thread.isMainThread, message)
-}
```

Comment:
We have no Shared/GeneralUtils for small, isolated utilties. And rather than create more files, I repurposed this one.